src/states/node.rs

```diff
@@ -1579,7 +1561,13 @@ impl Node {
                    route: u8,
                    sent_to: &[XorName])
                    -> Result<(Vec<XorName>, Vec<PeerId>), RoutingError> {
-        if self.is_proper() {
+        let force_via_proxy = match routing_msg.content {
```

Comment:
Won't this cause the sending to fail if we don't have a proxy anymore? E.g. when groups are merging, both of the connecting nodes could be without proxy.